### ISSUE TITLE
wrong version warning for qgis.core.QgsVectorLayer.fieldConfigurationFlags

### ISSUE THREAD

**Issue by pers-ephone** [qgis/pyqgis#138]
This part of the documentation states _New in version 3.16._ but the [qgis.core.Qgis.FieldConfigurationFlag](https://qgis.org/pyqgis/master/core/Qgis.html#qgis.core.Qgis.FieldConfigurationFlag) it returns is _New in version 3.34._

the version warning should be _New in version 3.34._

Page URL: https://qgis.org/pyqgis/master/core/QgsVectorLayer.html#qgis.core.QgsVectorLayer.fieldConfigurationFlags

**Comment by DelazJ** [comment]
@pers-ephone I think the method [already existed](https://qgis.org/pyqgis/3.16/core/QgsVectorLayer.html#qgis.core.QgsVectorLayer.setExcludeAttributesWfs) in 3.16 (the setter is mentioned but I can't find its own description in the docs). An oversight in the docs build?
I think https://github.com/qgis/QGIS/pull/54809 is the PR that updated and exposed the flags. Maybe @nyalldawson can shed some light?

**Comment by 3nids** [comment]
The flag has been moved from `core.QgsVectorLayer` to `core.QGIS`, that's why they are "new 3.34". 
The method in itself exists since 3.16, this is correct.

In Python, we do a monkey patching so that `FieldConfigurationFlags` exist in both places (`core.QgsVectorLayer` and `core.QGIS`).

To summarize, everything seems fine.

**Comment by pers-ephone** [comment]
> The method in itself exists since 3.16

Sorry to not have mentioned it before but the reason i knew of this issue was that the method does not exist in 3.28. From testing not much seems fine

From testing in 3.16.16 and in 3.28.12 (same results):
```py
# TLDR: Nothing exists
>>> qgis.core.QgsVectorLayer.fieldConfigurationFlags
AttributeError: type object 'QgsVectorLayer' has no attribute 'fieldConfigurationFlags'
>>> qgis.core.Qgis.fieldConfigurationFlags
AttributeError: type object 'Qgis' has no attribute 'fieldConfigurationFlags'

>>> qgis.core.QgsVectorLayer.FieldConfigurationFlags
AttributeError: type object 'QgsVectorLayer' has no attribute 'FieldConfigurationFlags'
>>> qgis.core.Qgis.FieldConfigurationFlags
AttributeError: type object 'Qgis' has no attribute 'FieldConfigurationFlags'

>>> qgis.core.QgsVectorLayer.FieldConfigurationFlag
AttributeError: type object 'QgsVectorLayer' has no attribute 'FieldConfigurationFlag'
>>> qgis.core.Qgis.FieldConfigurationFlag
AttributeError: type object 'Qgis' has no attribute 'FieldConfigurationFlag'

>>> qgis.core.QgsVectorLayer.setFieldConfigurationFlag
AttributeError: type object 'QgsVectorLayer' has no attribute 'setFieldConfigurationFlag'
>>> qgis.core.Qgis.setFieldConfigurationFlag
AttributeError: type object 'Qgis' has no attribute 'setFieldConfigurationFlag'
```

From testing in 3.34.0:
```py
>>> qgis.core.QgsVectorLayer.fieldConfigurationFlags
<built-in function fieldConfigurationFlags>
>>> qgis.core.Qgis.fieldConfigurationFlags
AttributeError: type object 'Qgis' has no attribute 'fieldConfigurationFlags'

>>> qgis.core.QgsVectorLayer.FieldConfigurationFlags
AttributeError: type object 'QgsVectorLayer' has no attribute 'FieldConfigurationFlags'
>>> qgis.core.Qgis.FieldConfigurationFlags
<class 'qgis._core.Qgis.FieldConfigurationFlags'>

>>> qgis.core.QgsVectorLayer.FieldConfigurationFlag
AttributeError: type object 'QgsVectorLayer' has no attribute 'FieldConfigurationFlag'
>>> qgis.core.Qgis.FieldConfigurationFlag
<enum 'FieldConfigurationFlag'>

>>> qgis.core.QgsVectorLayer.setFieldConfigurationFlag
<built-in function setFieldConfigurationFlag>
>>> qgis.core.Qgis.setFieldConfigurationFlag
AttributeError: type object 'Qgis' has no attribute 'setFieldConfigurationFlag'
```

**Comment by 3nids** [comment]
found it: https://github.com/qgis/QGIS/pull/54753

they were added later in the Pythin bindgings (3.34)
